sphinx_gallery/gen_rst.py

```diff
@@ -604,58 +600,113 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf):
         sys.argv[0] = src_file
         sys.argv[1:] = []
 
-    for blabel, bcontent, lineno in script_blocks:
-        if blabel == 'code':
-            code_output, rtime = execute_code_block(compiler, src_file,
-                                                    bcontent, lineno,
-                                                    example_globals,
-                                                    block_vars, gallery_conf)
+    t_start = time()
+    output_blocks = [execute_code_block(compiler, src_file, block,
+                                        example_globals,
+                                        block_vars, gallery_conf)
+                     for block in script_blocks]
+    time_elapsed = time() - t_start
+
+    sys.argv = argv_orig
+    clean_modules()
+
+    example_rst = rst_notebook_cell(script_blocks, output_blocks,
+                                    file_conf, gallery_conf)
+    save_rst_notebook(example_rst, example_file.replace('.py', '.rst'),
+                      time_elapsed, gallery_conf)
+
+    save_thumbnail(image_path_template, src_file, file_conf, gallery_conf)
+
+    example_nb = jupyter_notebook(script_blocks)
+    save_notebook(example_nb, example_file.replace('.py', '.ipynb'))
+
+    if block_vars['execute_script']:
+        logger.debug("%s ran in : %.2g seconds\n", src_file, time_elapsed)
+        # Writes md5 checksum if example has build correctly
+        # not failed and was initially meant to run(no-plot shall not cache
+        # md5sum)
+        with open(example_file + '.md5', 'w') as file_checksum:
+            file_checksum.write(get_md5sum(example_file))
+
+    return intro, time_elapsed
+
+
+def rst_notebook_cell(script_blocks, output_blocks, file_conf, gallery_conf):
```

Comment:
I think the name is not great. For me cell is similar to block. Here you are generating almost the full example rst unless I am missing something. Also notebook does not seem to be related either. In my mind, notebook only pertains to when you have multiple cells (see `is_example_notebook_like` below).